diff --git a/include/fmt/posix.h b/include/fmt/posix.h
index 35752a2..9e9d132 100644
--- a/include/fmt/posix.h
+++ b/include/fmt/posix.h
@@ -353,7 +353,9 @@ long getpagesize();
 
 #if (defined(LC_NUMERIC_MASK) || defined(_MSC_VER)) && \
     !defined(__ANDROID__) && !defined(__CYGWIN__) && !defined(__OpenBSD__)
-# define FMT_LOCALE
+# if !defined(__SWITCH__)
+#  define FMT_LOCALE
+# endif
 #endif
 
 #ifdef FMT_LOCALE
